app/src/services/sync.js

```diff
@@ -21,23 +21,34 @@ const service = {
    * @function _deriveObjectId
    * Checks an S3 Object for any previous `coms-id` tag traces and returns it if found.
    * Otherwise it writes a new `coms-id` to the S3 Object if none were previously found.
+   * If the `coms-id` conflicts with an existing COMS object, it is replaced with a new one.
    * @param {object | boolean} s3Object The result of an S3 HeadObject operation
    * @param {string} path String representing the canonical path for the specified object
    * @param {string | null} bucketId uuid of bucket or `null` if syncing object in default bucket
-   * @returns {Promise<string>} Resolves to an existing objectId or creates a new one
+   * @returns {Promise<string>} Resolves to an existing objectId (if not already in COMS)
+   *                            or creates a new one
    */
   _deriveObjectId: async (s3Object, path, bucketId) => {
     let objId = uuidv4();
 
     if (typeof s3Object === 'object') { // If regular S3 Object
       const TagSet = await storageService.getObjectTagging({ filePath: path, bucketId: bucketId })
         .then(result => result.TagSet ?? []);
+
       const s3ObjectComsId = TagSet.find(obj => (obj.Key === 'coms-id'))?.Value;
 
-      if (s3ObjectComsId && uuidValidate(s3ObjectComsId)) {
+      if (s3ObjectComsId
+        && uuidValidate(s3ObjectComsId)
+        && !(await objectService.exists(s3ObjectComsId))) {
+        // re-use existing coms-id (if no conflict)
         objId = s3ObjectComsId;
-      } else { // Update S3 Object if there is still remaining space in the TagSet
-        if (TagSet.length < 10) { // putObjectTagging replaces S3 tags so new TagSet must contain existing values
+      } else {
+        // remove `coms-id` tag since it conflicts with an existing COMS object
+        TagSet.filter(x => x.Key != 'coms-id');
```

Comment:
i think `TagSet.filter()` creates a copy but wont change the value of `TagSet`. you need to re-assign it. (eg: `const filtered = Tagse..`) and then use that const in the code below.
or you could change TagSet with: `TagSet.splice(TagSet.findIndex(e => e.Key=== "coms-id"),1);`

Reply:
Good catch - I've tweaked the`TagSet` declaration a little further up (it's now a variable instead of a constant) and made this line assign the filtered results (instead of just `filter()`'ing it and calling it a day).